include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -375,14 +382,16 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 			// Alternatively, adding the time an SEntry entered the cache could also serve this purpose
 			inline void merge(const CCache* other) {
 				for (auto& entry : other->m_container) {
-					m_container.insert(entry);
+					SEntry entryCopy(entry);
+					m_container.insert(entryCopy);
```

Comment:
you end up with 2 copies here, better approach would be to `std::move(entryCopy)`, best would be to use `emplace(entry)` which forwards the arguments to a constructor and does a zero copy initialization

Reply:
same thing happens later on

Reply:
Right, I thought I had put a move here, maybe I lost it in the refactor. I'll try the best approach or eventually the netter one if I can't make the best one work easily

Reply:
yeah if you can't figure it out in 30 minutes, do the second best thing